src/ReCaptchaResponse.php

```diff
@@ -0,0 +1,271 @@
+
+namespace Google\ReCaptcha;
+
+use Serializable;
+use JsonSerializable;
+
+/**
+ * Class ReCaptchaResponse
+ *
+ * @package Google\ReCaptcha\Http
+ *
+ * @property-read null|string $hostname
+ * @property-read null|string $challenge_ts
+ * @property-read null|string $apk_package_name
+ * @property-read null|float $score
+ * @property-read null|string $action
+ * @property-read array $error_codes
+ * @property-read bool $success
+ */
+class ReCaptchaResponse implements JsonSerializable, Serializable
+{
+    /**
+     * ReCaptchaResponse attributes.
+     *
+     * @var array
+     */
+    protected $attributes = [];
+
+    /**
+     * Additional constraints to check after the response is received.
+     *
+     * @var array
+     */
+    protected $constraints = ReCaptcha::CONSTRAINTS_ARRAY;
+
+    /**
+     * ReCaptchaResponse constructor.
+     *
+     * @param  array $attributes
+     */
+    public function __construct(array $attributes = [])
+    {
+        $this->attributes = $attributes;
+    }
+
+    /**
+     * Returns all the response attributes as an array
+     *
+     * @return array
+     */
+    public function getAttributes()
+    {
+        return $this->attributes;
+    }
+
+    /**
+     * Returns if the challenge is valid.
+     *
+     * @return bool
+     */
+    public function valid()
+    {
+        return ($this->attributes['success'] ?? false) && $this->error_codes === [];
+    }
+
+    /**
+     * Returns if the challenge is invalid.
+     *
+     * @return bool
+     */
+    public function invalid()
+    {
+        return ! $this->valid();
+    }
+
+    /**
+     * Return if the response has a particular error
+     *
+     * @param  mixed ...$errors
+     * @return bool
+     */
+    public function hasError(...$errors)
+    {
+        if (empty($errors)) {
+            return false;
+        }
+
+        $array = $this->error_codes ?? [];
+
+        foreach ($errors as $error) {
+            if (! in_array($error, $array, true)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Add ann array of errors to the existing error codes.
+     *
+     * @param  array $errors
+     * @return \Google\ReCaptcha\ReCaptchaResponse
+     */
+    public function addErrors(array $errors = [])
+    {
+        $this->attributes['error-codes'] = array_merge($this->attributes['error-codes'] ?? [], $errors);
+
+        return $this;
+    }
+
+    /**
+     * Return a single constraint value.
+     *
+     * @param  string $name
+     * @return mixed
+     */
+    public function constraint(string $name)
+    {
+        return $this->constraints[$name];
+    }
+
+    /**
+     * Returns the constraints to check this response.
+     *
+     * @return array
+     */
+    public function constraints()
+    {
+        return $this->constraints;
+    }
+
+    /**
+     * Sets the constraints to check this response.
+     *
+     * @param  array $constraints
+     * @return $this
```

Comment:
`self` seemed a better practice

Reply:
I usually use this:
* `$this` if its the same instance for a dynamic method (like fluent/chained methods).
* `self` only for interfaces.
* `static` for static methods or when returning a new instance.

Also, there is no PHPDoc documentation of what on when, AFAIK.

Reply:
https://docs.phpdoc.org/latest/references/phpdoc/tags/return.html
"When provided it MUST contain a Type to indicate what is returned"
https://docs.phpdoc.org/latest/references/phpdoc/types.html
```
keyword                  = "string"|"integer"|"int"|"boolean"|"bool"|"float"
                           |"double"|"object"|"mixed"|"array"|"resource"|"scalar"
                           |"void"|"null"|"callback"|"false"|"true"|"self"
```

So maybe replace $this ?